docs/data/tree-view/overview/CustomizedTreeView.tsx

```diff
@@ -1,105 +1,190 @@
-    </SvgIcon>
-  );
+declare module 'react' {
+  interface CSSProperties {
+    '--tree-view-color'?: string;
+    '--tree-view-bg-color'?: string;
+  }
 }
 
-function CloseSquare(props: SvgIconProps) {
-  return (
-    <SvgIcon
-      className="close"
-      fontSize="inherit"
-      style={{ width: 14, height: 14 }}
-      {...props}
-    >
-      {/* tslint:disable-next-line: max-line-length */}
-      <path d="M17.485 17.512q-.281.281-.682.281t-.696-.268l-4.12-4.147-4.12 4.147q-.294.268-.696.268t-.682-.281-.281-.682.294-.669l4.12-4.147-4.12-4.147q-.294-.268-.294-.669t.281-.682.682-.281.696 .268l4.12 4.147 4.12-4.147q.294-.268.696-.268t.682.281 .281.669-.294.682l-4.12 4.147 4.12 4.147q.294.268 .294.669t-.281.682zM22.047 22.074v0 0-20.147 0h-20.12v0 20.147 0h20.12zM22.047 24h-20.12q-.803 0-1.365-.562t-.562-1.365v-20.147q0-.776.562-1.351t1.365-.575h20.147q.776 0 1.351.575t.575 1.351v20.147q0 .803-.575 1.365t-1.378.562v0z" />
-    </SvgIcon>
-  );
-}
+type StyledTreeItemProps = TreeItemProps & {
+  labelIcon: React.ElementType;
+  labelText: string;
+};
+
+const StyledTreeItemLabel = styled(Typography)({
+  color: 'inherit',
+  fontFamily: 'General Sans',
+  fontWeight: 'inherit',
+  flexGrow: 1,
+}) as unknown as typeof Typography;
+
+const StyledTreeItemRoot = styled(TreeItem)(({ theme }) => ({
+  color:
+    theme.palette.mode === 'light'
+      ? theme.palette.grey[800]
+      : theme.palette.grey[400],
+  position: 'relative',
+  [`& .${treeItemClasses.content}`]: {
+    flexDirection: 'row-reverse',
+    borderRadius: theme.spacing(0.7),
+    marginBottom: theme.spacing(0.5),
+    marginTop: theme.spacing(0.5),
+    padding: theme.spacing(0.5),
+    paddingRight: theme.spacing(1),
+    fontWeight: 500,
+    [`& .${treeItemClasses.label}`]: {
+      fontWeight: 'inherit',
+    },
+    [`& .${treeItemClasses.iconContainer}`]: {
+      marginRight: theme.spacing(2),
+    },
+    [`&.Mui-expanded `]: {
+      '&:not(.Mui-focused, .Mui-selected, .Mui-selected.Mui-focused) .labelIcon': {
+        color: '#4b5aff',
+      },
+      '&::before': {
+        content: '""',
+        display: 'block',
+        position: 'absolute',
+        left: '16px',
+        top: '44px',
+        height: 'calc(100% - 48px)',
+        width: '1.5px',
+        backgroundColor:
+          theme.palette.mode === 'light'
+            ? theme.palette.grey[300]
+            : theme.palette.grey[700],
+      },
+    },
+    '&:hover': {
+      // backgroundColor: '#f0f2ff',
+      backgroundColor: alpha('#4b5aff', 0.1),
+      color: theme.palette.mode === 'light' ? '#4b5aff' : 'white',
+    },
+    [`&.Mui-focused, &.Mui-selected, &.Mui-selected.Mui-focused`]: {
+      backgroundColor: '#4b5aff',
```

Comment:
What do you think of using a value from the theme here (and in any other case where we're using arbitrary hex values)? Just so we maintain the consistency given certain colors are already being picked up from there!

Reply:
Went for the primary color from the theme. I enjoyed the purple-ish color, but using the theme makes sense for consistency